subprojects/core/src/main/java/org/gradle/internal/taskgraph/PlannedTask.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.taskgraph;
+
+import org.gradle.api.internal.project.taskfactory.TaskIdentity;
+import org.gradle.internal.scan.UsedByScanPlugin;
+
+import java.util.List;
+
+@UsedByScanPlugin
+public interface PlannedTask {
+
+    TaskIdentity getTask();
+
+    List<TaskIdentity> getDependencies();
+
+    List<TaskIdentity> getMustRunAfter();
+
+    List<TaskIdentity> getShouldRunAfter();
+
+    List<TaskIdentity> getFinalizedBy();
```

Comment:
This means that `TaskIdentity` is now also used by the scan plugin. Shouldn't there be some intermediate interface, or just a plain `String`, to decouple the scan plugin from the internal `TaskIdentity`?

Reply:
introduced ExecuteTaskBuildOperationType.TaskIdentity as discussed